tensorflow_addons/image/filters_test.py

```diff
@@ -23,73 +23,129 @@
+    def test_invalid_padding(self):
+        msg = ("padding should be one of \"REFLECT\", \"CONSTANT\", "
+               "or \"SYMMETRIC\".")
+        image = tf.ones(shape=(1, 28, 28, 1))
 
-    @test_utils.run_deprecated_v1
-    def test_dimension(self):
-        for image_shape in [(3, 4, None), (3, None, 4), (None, 3, 4)]:
-            with self.assertRaises(TypeError):
-                tf_img = tf.compat.v1.placeholder(tf.int32, shape=image_shape)
-                mean_filter2d(tf_img)
-
-    @test_utils.run_in_graph_and_eager_modes
-    def test_image_vs_filter(self):
-        tf_img = tf.zeros([3, 4, 3], tf.int32)
-        filter_shape = (3, 5)
-        with self.assertRaises(ValueError):
-            mean_filter2d(tf_img, filter_shape)
-
-    @test_utils.run_in_graph_and_eager_modes
-    def test_three_channels(self):
-        tf_img = [[[0.32801723, 0.08863795, 0.79119259],
-                   [0.35526001, 0.79388736, 0.55435993],
-                   [0.11607035, 0.55673079, 0.99473371]],
-                  [[0.53240645, 0.74684819, 0.33700031],
-                   [0.01760473, 0.28181609, 0.9751476],
-                   [0.01605137, 0.8292904, 0.56405609]],
-                  [[0.57215374, 0.10155051, 0.64836128],
-                   [0.36533048, 0.91401874, 0.02524159],
-                   [0.56379134, 0.9028874, 0.19505117]]]
-
-        tf_img = tf.convert_to_tensor(value=tf_img)
-        expt = [[[34, 54, 75], [38, 93, 119], [14, 69, 87]],
-                [[61, 82, 94], [81, 147, 144], [40, 121, 93]],
-                [[42, 57, 56], [58, 106, 77], [27, 82, 49]]]
-        expt = tf.convert_to_tensor(value=expt)
-        self._validate_mean_filter2d(tf_img, expt)
+        with self.assertRaisesRegexp(ValueError, msg):
+            mean_filter2d(image, padding="TEST")
+
+    def test_3d_image(self):
+        # Test shape (3, 3, 1)
+        image = tf.constant([[1., 2., 3.], [4., 5., 6.], [7., 8., 9.]],
+                            shape=(3, 3, 1))
+
+        # (3, 3) filter shape:
+        # reflected padding with 1 pixel to each direction
+        #   5 4 5 6 5
+        #   2 1 2 3 2
+        #   5 4 5 6 5
+        #   8 7 8 9 8
+        #   5 4 5 6 5
+
+        expected = tf.constant(
+            [[33. / 9., 36. / 9., 39. / 9.], [42. / 9., 45. / 9., 48. / 9.],
+             [51. / 9., 54. / 9., 57. / 9.]],
+            shape=(3, 3, 1))
+
+        output = mean_filter2d(image)
+        self.assertAllClose(output, expected)
+
+        # Test shape (3, 3, 3)
+        image = tf.stack([image, 2. * image, 3. * image], axis=-1)
+        expected = tf.stack([expected, 2. * expected, 3. * expected], axis=-1)
+        # Squeeze shape from (3, 3, 1, 3) to (3, 3, 3)
+        image = tf.squeeze(image, axis=-2)
+        expected = tf.squeeze(expected, axis=-2)
+
+        output = mean_filter2d(image)
+        self.assertAllClose(output, expected)
+
+    def test_4d_image(self):
+        # Test shape (2, 3, 3, 1)
```

Comment:
Can we add more unit tests, say:
1. channes_num is 3
2. uint8 images
3. float64 images